flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/CheckpointCoordinator.java

```diff
@@ -648,24 +636,17 @@ public boolean isShutdown() {
 			return checkpoint.getCompletionFuture();
 		}
 		catch (Throwable t) {
-			// guard the map against concurrent modifications
-			synchronized (lock) {
-				pendingCheckpoints.remove(checkpointID);
-			}
-
 			int numUnsuccessful = numUnsuccessfulCheckpointsTriggers.incrementAndGet();
 			LOG.warn("Failed to trigger checkpoint {} for job {}. ({} consecutive failed attempts so far)",
 					checkpointID, job, numUnsuccessful, t);
 
-			if (!checkpoint.isDiscarded()) {
-				failPendingCheckpoint(checkpoint, CheckpointFailureReason.TRIGGER_CHECKPOINT_FAILURE, t);
-			}
-
-			try {
-				checkpointStorageLocation.disposeOnFailure();
```

Comment:
What has happened to this line? Was it replaced by something?

Reply:
It's duplicated. `abortPendingCheckpoint` -> `PendingCheckpoint#dispose` will invoke `CheckpointStorageLocation#disposeOnFailure`. That's the benefit of unifying error handling into `abortPendingCheckpoint`. We don't have to release resource everywhere.